File: Queues/ExerciseBuildingAQueueUsingArrays/ArrayQueue.py
class ArrayQueue:
    def __init__(self):
        self.__elements = []

    def enqueue(self, item):
        self.__elements.append(item)

    def dequeue(self):
        if (len(self.__elements) == 0):
            raise Exception('The Queue is empty')

        return self.__elements.pop(0)

    def peek(self):
        if (len(self.__elements) == 0):
            raise Exception('The Queue is empty')

        return self.__elements[0]

    def isEmpty(self):
        return (len(self.__elements) == 0)

    def getQueue(self):
        return self.__elements

File: Queues/QueueExercises/ArrayQueue.js
const _elements = new WeakMap();

class ArrayQueue{
    constructor(){
        _elements.set(this, []);
    }

    enqueue(item){
       return _elements.get(this).push(item)
    }

    dequeue(){
        if (_elements.get(this).length==0) throw new Error('Queue is empty');
        return _elements.get(this).shift();
    }

    peek(){
        if (_elements.get(this).length==0) throw new Error('Queue is empty');
        return _elements.get(this)[0];
    }

    isEmpty(){
        return (_elements.get(this).length==0);
    }

    getQueue(){
        return _elements.get(this);
    }

    length(){
        return _elements.get(this).length;
    }
}

module.exports = ArrayQueue;